c8/connection.py

```diff
@@ -77,14 +79,26 @@ def __init__(self, url, email, password, token, apikey, http_client):
 
         # TODO : Handle the functions side of things
 
-    def _get_auth_token(self):
-        data = {
-            "email": self._email,
-            "password": self._password,
-        }
+    def _get_auth_token(self, payload=None):
+        data = {}
+
+        if payload is not None:
+            if "email" in payload and "password" in payload:
```

Comment:
All the following statements inside "if payload is not None" should not be needed, you should simply be able to do data = payload that's it.